vllm/attention/ops/prefix_prefill.py

```diff
@@ -675,24 +696,49 @@ def context_attention_fwd(q,
                               k,
                               v,
                               o,
+                              kv_cache_dtype: str,
                               k_cache,
                               v_cache,
                               b_loc,
                               b_start_loc,
                               b_seq_len,
                               b_ctx_len,
                               max_input_len,
+                              fp8_k_scale: float = 1.0,
+                              fp8_v_scale: float = 1.0,
                               alibi_slopes=None,
                               sliding_window=None):
 
         cap = current_platform.get_device_capability()
         BLOCK = 128 if cap[0] >= 8 else 64
+        NUM_WARPS = 8
 
         # need to reduce num. blocks when using fp32
         # due to increased use of GPU shared memory
         if q.dtype is torch.float32:
             BLOCK = BLOCK // 2
 
+        # Conversion of FP8 Tensor from uint8 storage to
+        # appropriate torch.dtype for interpretation by Triton
+        if "fp8" in kv_cache_dtype:
+            assert (k_cache.dtype == torch.uint8)
+            assert (v_cache.dtype == torch.uint8)
+
+            if kv_cache_dtype == "fp8" or kv_cache_dtype == "fp8_e4m3":
+                target_dtype = torch.float8_e4m3fn
+            elif kv_cache_dtype == "fp8_e5m2":
+                target_dtype = torch.float8_e5m2
+            else:
+                raise ValueError("Unsupported FP8 dtype:", kv_cache_dtype)
+
+            k_cache = k_cache.view(target_dtype)
+            v_cache = v_cache.view(target_dtype)
+
+        if (k_cache.dtype == torch.uint8
+                or v_cache.dtype == torch.uint8 and kv_cache_dtype == "auto"):
+            raise ValueError("kv_cache_dtype='auto' unsupported for\
+                FP8 KV Cache prefill kernel")
```

Comment:
Blocked by #7302

Reply:
Yes we can land this first and get rid of it in #7302 depending on merge order

Reply:
@comaniac I think #7302 is more controversial; prefer to merge this PR first if everything checks out. WDYT?

Follow up upon merge is post issue about the illegal memory access

Specifically one of these errors:

```python
RuntimeError: Triton Error [CUDA]: an illegal memory access was encountered
RuntimeError: CUDA error: an illegal memory access was encountered
```

Reply:
I closed #7302 by virtue of being unnecessary complexity at no benefit. This last-mile dispatch is fine IMO